tests/base.py

```diff
@@ -0,0 +1,377 @@
+    REPLICATION_METHOD = "forced-replication-method"
+    API_LIMIT = "max-row-limit"
+    INCREMENTAL = "INCREMENTAL"
+    FULL_TABLE = "FULL_TABLE"
+    START_DATE_FORMAT = "%Y-%m-%dT00:00:00Z"
+    BOOKMARK_COMPARISON_FORMAT = "%Y-%m-%dT00:00:00.000000Z"
+
+    LOGGER = singer.get_logger()
+
+    start_date = ""
+
+    @staticmethod
+    def tap_name():
+        """The name of the tap"""
+        return "tap-facebook"
+
+    @staticmethod
+    def get_type():
+        """the expected url route ending"""
+        return "platform.facebook"
+
+    def get_properties(self, original: bool = True):
+        """Configuration properties required for the tap."""
+        return_value = {
+            'start_date' : '2015-03-15T00:00:00Z',
+            'account_id': os.getenv('TAP_FACEBOOK_ACCOUNT_ID'),
+            'end_date': '2015-03-16T00:00:00+00:00',
+            'insights_buffer_days': '1'
+        }
+        if original:
+            return return_value
+
+        # This test needs the new connections start date to be larger than the default
+        assert self.start_date > return_value["start_date"]
+
+        return_value["start_date"] = self.start_date
+        return return_value
+
+    @staticmethod
+    def get_credentials():
+        """Authentication information for the test account"""
+        return {'access_token': os.getenv('TAP_FACEBOOK_ACCESS_TOKEN')}
+
+    def expected_metadata(self):
+        """The expected streams and metadata about the streams"""
+        return {
+            "ads": {
+                self.PRIMARY_KEYS: {"id", "updated_time"},
+                self.REPLICATION_METHOD: self.INCREMENTAL,
+                self.REPLICATION_KEYS: {"updated_time"}
+            },
+            "adcreative": {
+                self.PRIMARY_KEYS: {"id"},
+                self.REPLICATION_METHOD: self.FULL_TABLE,
+            },
+            "adsets": {
+                self.PRIMARY_KEYS: {"id", "updated_time"},
+                self.REPLICATION_METHOD: self.INCREMENTAL,
+                self.REPLICATION_KEYS: {"updated_time"}
+            },
+            "campaigns": {
+                self.PRIMARY_KEYS: {"id", },
+                self.REPLICATION_METHOD: self.INCREMENTAL,
+                self.REPLICATION_KEYS: {"updated_time"}
+            },
+            "ads_insights": {
+                self.PRIMARY_KEYS: {"campaign_id", "adset_id", "ad_id", "date_start"},
+                self.REPLICATION_METHOD: self.INCREMENTAL,
+                self.REPLICATION_KEYS: {"date_start"}
+            },
+            "ads_insights_age_and_gender": {
+                self.PRIMARY_KEYS: {
+                    "campaign_id", "adset_id", "ad_id", "date_start", "age", "gender"
+                },
+                self.REPLICATION_METHOD: self.INCREMENTAL,
+                self.REPLICATION_KEYS: {"date_start"}
+            },
+            "ads_insights_country": {  # TODO - BUG ? country is listed in stitch docs, but not considerd a pk by the tap
```

Comment:
Yep, there's a backlog ticket for this https://stitchdata.atlassian.net/browse/SRCE-2555

Reply:
I'll link in the test, thanks